pkg/controllers/user/nslabels/labels.go

```diff
@@ -0,0 +1,80 @@
+package nslabels
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/rancher/norman/clientbase"
+	typescorev1 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+const (
+	projectIDField = "field.cattle.io/projectId"
+)
+
+type namespaceHandler struct {
+	nsLister typescorev1.NamespaceLister
+	nsClient *clientbase.ObjectClient
+}
+
+func Register(cluster *config.UserContext) {
+	logrus.Infof("Registering namespaceHandler for adding labels ")
+	nsh := &namespaceHandler{
+		cluster.Management.Core.Namespaces("").Controller().Lister(),
+		cluster.Core.Namespaces("").ObjectClient(),
+	}
+	cluster.Core.Namespaces("").AddHandler("namespaceHandler", nsh.Sync)
+}
+
+func (nsh *namespaceHandler) Sync(key string, ns *corev1.Namespace) error {
+	if ns == nil {
+		return nil
+	}
+	logrus.Infof("namespaceHandler Sync key=%v, ns=%+v", key, *ns)
```

Comment:
This is going to log too much.  You should really only log if you actually perform an action.  Sync handlers are called for ever change and then also on an interval.

Reply:
Ok, will move it to debug